### PR TITLE
Add support for `-l` to pwsh so that it is compatible with POSIX shell expectations

### LINKED ISSUE #3600: Support `-l` for pwsh.exe
I hit a problem with vscode integrated terminal after changing my default shell to powershell on macOS.

https://github.com/Microsoft/vscode/issues/25035

I don't know how common is it to use `-l` as `-noprofile` in other shells, but if it's a fairly common interface we may want to support the same in ps.

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

POSIX shells have a `-l` switch to indicate it is a login shell which will process profiles.  However, PowerShell by default already loads the profile and requires `-noprofile` to explicitly not load it.  To make pwsh compatible with other tools (like VSCode) that expects a `-l` switch, the change is to add `-loadprofile` aliased as `-l` so that it doesn't complain about that switch which is a no-op since we load the profile by default.

Also sorted the switches alphabetically in the code.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/3600

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] https://github.com/MicrosoftDocs/PowerShell-Docs/pull/4261
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs [hunk 1]
```diff
@@ -173,28 +173,29 @@ internal class CommandLineParameterParser
         private const int MaxPipePathLengthMacOS = 104;
 
         internal static string[] validParameters = {
-            "version",
-            "nologo",
-            "noexit",
 #if STAMODE
             "sta",
             "mta",
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Not related the PR - we'll need to review apartments after moving to .Net Core 3.0.

Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Not related the PR - we'll need to review apartments after moving to .Net Core 3.0.

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 2]
```diff
@@ -245,6 +245,33 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
         }
     }
 
+    Context "-LoadProfile Commandline switch" {
+        BeforeAll {
+            if (Test-Path $profile) {
```
Comment (iSazonov) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: It seems `$profile` is null on CIs.
Also can we change the file on containers or it is read-only?

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: https://github.com/PowerShell/PowerShell/issues/7216

File: src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx [hunk 3]
```diff
@@ -224,6 +224,9 @@ All parameters are case-insensitive.</value>
 -Interactive | -i
     Present an interactive prompt to the user. Inverse for NonInteractive parameter.
 
+-LoadProfile | -l
+    Load the PowerShell profiles.  This is the default behavior even if this is not specified.
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx]:
```suggestion
    Load the PowerShell profiles. This is the default behavior even if this is not specified.
```

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
Based on new RFC I think this should be `-Login` or `-LoginShell` parameter.

**Comment by SteveL-MSFT** [comment]
@iSazonov the problem with calling it `-LoginShell` is that it doesn't make any sense for Windows.  I don't think Linux users care about the long form as they would just use `-l`.

**Comment by iSazonov** [comment]
Linux users will very confused to see long form in docs - they expect "login" but get another. The current solution looks like a hoax. I think we could discuss this in your RFC.

**Comment by SteveL-MSFT** [comment]
@iSazonov bring up as an issue in the RFC, we can still change while in Preview

**Comment by TravisEz13** [comment]
@anmenaga Please be sure to use squash and merge unless there is a good reason.

**Pull request merged**